recorder/src/main/c/preload.c

```diff
@@ -0,0 +1,42 @@
+#define _GNU_SOURCE
+#include <pthread.h>
+#include <signal.h>
+#include <assert.h>
+#include <stdio.h>
+#include <jni.h>
+#include <dlfcn.h>
+
+int JLI_Launch(int argc, char ** argv,
+           int jargc, const char** jargv,
+           int appclassc, const char** appclassv,
+           const char* fullversion,
+           const char* dotversion,
+           const char* pname,
+           const char* lname,
+           jboolean javaargs,
+           jboolean cpwildcard,
+           jboolean javaw,
+           jint ergo
+    ) {
+
+    sigset_t set;
+    assert(sigemptyset(&set) == 0);
+    assert(sigaddset(&set, SIGPROF) == 0);
+    assert(pthread_sigmask(SIG_BLOCK, &set, NULL) == 0);
```

Comment:
The main fn (java ...) calls jli-launcher, which dispatches vm-creation on a new thread. Turns out a lot of sigprof invocations end up getting handled on group-leader (pid, not random(tid)). So this is a hack to suppress that. Its a thread that waits on the vm-thread join till the very end, so will have no CPU burn anyway, so its a safe bet.

This will improve our data-quality a lot.